packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -122,7 +122,7 @@ export function useGridDimensions(
       };
 
       const scrollInformation = hasScroll({
-        content: { width: Math.round(columnsTotalWidth), height: rowsMeta.currentPageTotalHeight },
+        content: { width: columnsTotalWidth, height: rowsMeta.currentPageTotalHeight },
```

Comment:
In https://github.com/mui/mui-x/pull/9236 we started to use fractional size. Rounding the columns total width breaks the equality comparison